habitat-lab/habitat/datasets/rearrange/samplers/object_sampler.py

```diff
@@ -261,6 +261,12 @@ def sample_placement(
                     new_object.rotation = (
                         habitat_sim.utils.common.random_quaternion()
                     )
+            new_object_height = (
+                new_object.root_scene_node.cumulative_bb.size()[1]
+            )
+            new_object.translation = new_object.translation + mn.Vector3(
```

Comment:
For  AABB receptacles, we have upper bounds on the max height the object is allowed to take. (example - for a drawer, these would correspond to the bounding box). There is a chance that the object can exceed the bounds with this addition. This could result in the object being placed on a receptacle surface that is above the intended receptacle. Maybe you can add a check: for AABBReceptacle check if the object is within the bounds, or we can make change to sample positions taking the object height into account.

Reply:
I've changed this to do the shift only for mesh receptacles.